autosklearn/pipeline/components/data_preprocessing/imputation/categorical_imputation.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+from sklearn.utils import check_array
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+from autosklearn.pipeline.components.base import AutoSklearnPreprocessingAlgorithm
+from autosklearn.pipeline.constants import DENSE, SPARSE, UNSIGNED_DATA, INPUT
+
+
+class CategoricalImputation(AutoSklearnPreprocessingAlgorithm):
+    """
+    Substitute missing values by 2
+    """
+
+    def __init__(self, random_state=None):
+        self.random_stated = random_state
+
+    def fit(self, X, y=None):
+        import sklearn.impute
+
+        self.preprocessor = sklearn.impute.SimpleImputer(
+            strategy='constant', fill_value=2, copy=False)
+        self.preprocessor = self.preprocessor.fit(X)
+        return self
+
+    def transform(self, X):
+        if self.preprocessor is None:
+            raise NotImplementedError()
+        X = self.preprocessor.transform(X)
+        X = check_array(X, accept_sparse='csc', dtype=np.int32)
```

Comment:
It is there basically to just convert X to integer. it had more functions in the past, but the code has changed and this line was kind of left behind. So I removed it and used instead a direct numpy conversion to integer. Also, I added a check in the category_shift (first component of the pipeline) to make sure that X is 100% numerical. So a cast after impuation won't fail. (Note that we have to wait until the imputation is done to make the cast, otherwise we would loose the np.nans)